### PR TITLE
Add `Dropout` and `train_ch3()`

### PR DESCRIPTION
**Pull request by aksub99**
Add Dropout from Chapter 4 of the book.
Also added `train_ch3()`  in `d2l/train.py` as in #4

### CODE REVIEW

File: d2l/train.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+
+def evaluate_accuracy(data_iter, net):
+    """Evaluate accuracy of a model on the given data set."""
+    acc_sum, n = torch.Tensor([0]), 0
+    for batch in data_iter:
+        features, labels = batch
+        for X, y in zip(features, labels):
+            y = y.type(torch.FloatTensor)
+            acc_sum += torch.sum((torch.argmax(net(X), dim=1).type(torch.FloatTensor) == y).data).float()
```
Comment (AnirudhDagar) [inline d2l/train.py]: I think It is a better practice to use `.item` instead of `.data` in this context in PyTorch.
`.data` will be deprecated in the newer versions. 
Please make the changes replacing .item in all the following lines.

Comment (aksub99) [reply d2l/train.py]: `.item()` can only be used to convert single element tensors to scalars. So, I've used `.detach()` for the multi-element cases.

**Comment by aksub99** [comment]
Please note that I've run the notebook again to update the outputs.

File: d2l/train.py (hunk 1 above)
Comment (AnirudhDagar) [reply d2l/train.py]: Yeah!! The loss will be a scalar always. Looks good now.

**Comment by aksub99** [comment]
Have squashed the commits. Is it ready to be merged?

**Comment by AnirudhDagar** [comment]
Thanks! 👍
